instrumentation/aws-sdk/src/main/java/software/amazon/opentelemetry/javaagent/instrumentation/awssdk_v1_11/AdotAwsSdkTracingRequestHandler.java

```diff
@@ -52,5 +73,21 @@ public void beforeRequest(Request<?> request) {}
    *     href="https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/aws-sdk/aws-sdk-1.11/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v1_11/TracingRequestHandler.java#L131">reference</a>
    */
   @Override
-  public void afterAttempt(HandlerAfterAttemptContext context) {}
+  public void afterAttempt(HandlerAfterAttemptContext context) {
+    Span currentSpan = Span.current();
+
+    if (currentSpan != null && currentSpan.getSpanContext().isValid()) {
+      Request<?> request = context.getRequest();
+      Response<?> response = context.getResponse();
+      Exception exception = context.getException();
+
+      AttributesBuilder attributes = Attributes.builder();
+      experimentalAttributesExtractor.onEnd(
+          attributes, Context.current(), request, response, exception);
+
+      attributes
+          .build()
+          .forEach((key, value) -> currentSpan.setAttribute(key.getKey(), value.toString()));
```

Comment:
The syntax is different from the previous: 
```
(key, value) -> currentSpan.setAttribute((AttributeKey<String>) key, (String) value)
```
Could you explain why this is different? Can we assume that the key and value are always strings?